### PR TITLE
Lmangall/web 168 implement automatic or manual socket fd closure before

### PR DESCRIPTION
**Pull request by lmangall**
I think we still don't have a way to test this, so we can pull it to main in the meantime if it looks fine to you

### CODE REVIEW

File: src/CGIHandler.cpp [hunk 1]
```diff
@@ -97,6 +98,17 @@ HTTPResponse CGIHandler::CGIStringToResponse(const std::string &cgiOutput)
 	return response;
 }
 
+void CGIHandler::closeAllSocketFDs()
+{
+	for (std::vector<struct pollfd>::iterator it = _FDsRef->begin(); it != _FDsRef->end(); ++it)
+	{
+		if (it->fd != _pollFd->fd)
```
Comment (552020) [inline src/CGIHandler.cpp:105]: This means that you are not closing the one used by the parent process. Why? We discussed this already I guess ...

**Review by 552020 — CHANGES_REQUESTED** [review]

**Comment by 552020** [comment]
+ conflicts.

File: src/CGIHandler.cpp (hunk 1 above)
Comment (lmangall) [reply src/CGIHandler.cpp]: Done

**Review by dantol29 — APPROVED** [review]
Looks good to me! Would be nice to save child process fd somewhere to make CGI non-blocking

**Review by 552020 — APPROVED** [review]
LRGTM!

**Pull request merged**